core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -639,19 +640,22 @@ private[spark] object Utils extends Logging {
       fs: FileSystem,
       conf: SparkConf,
       hadoopConf: Configuration,
-      fileOverwrite: Boolean): Unit = {
-    if (!targetDir.mkdir()) {
+      fileOverwrite: Boolean,
+      filename: Option[String] = None): Unit = {
```

Comment:
You're right... though that pretty unintuitive. We should clean this up separately after the release.